File: 프로그래머스/1/17681. ［1차］ 비밀지도/［1차］ 비밀지도.cpp
#include <string>
#include <vector>
#include <iostream>
#include <string>
using namespace std;


string binary(int num, int n){
    string s = "";
    
    while (num){
        char c = (num % 2) ? '#': ' '; 
        s = c + s; 
        num /= 2;
    }
    if (s.size() == n)
        return s;
    s.insert(0, n - s.size() , ' ');
    return s;
}

vector<string> solution(int n, vector<int> arr1, vector<int> arr2){
    vector<string> answer(arr1.size());
    // cout << binary(2);
    
    for (int i = 0; i < arr1.size(); i++){
        arr1[i] |= arr2[i]; 
        answer[i] = binary(arr1[i], arr1.size());
    }
    
    return answer;
}

/*
32 16 8 4 2 1
0  1  1 1 1 0
*/

File: 프로그래머스/1/17681. ［1차］ 비밀지도/［1차］ 비밀지도.py
def solution(n, arr1, arr2):
    answer = ["" for _ in range(n)]
    
    for i in range(n):
        bit = arr1[i] | arr2[i]
        bit = bin(bit).lstrip('0b').zfill(n)

        for b in bit:
            if b == "1":
                answer[i] += "#"
            else:
                answer[i] += " "
    
    return answer